test/system/autocompletes_test.rb

```diff
@@ -19,7 +19,7 @@ class AutocompletesTest < ApplicationSystemTestCase
   def wait_for_ajax
     Timeout.timeout(Capybara.default_max_wait_time) do
       loop do
-        active = page.evaluate_script("jQuery.active")
+        active = page.evaluate_script("if (jQuery !== undefined) { jQuery.active }")
```

Comment:
`Selenium::WebDriver::Error::JavascriptError: javascript error: Unexpected token 'if'`

Reply:
hmm. This is a little annoying because accessing jQuery on window is not safe with imports since it may not be loaded yet. I'm struggling a bit here with how to check if it's ready.

Reply:
ok, I've established without a doubt that selenium is not running the javascript for autocomplete. The content security policy is blocking all local javascript files in test.

The urls during test are like: `http://localhost:31337/assets/application-525f2ca7a98c72861092d17a0cb1e9e976154d231fae18ebf534d9a4a813bdf0.js`

It seems like the only solution is to allow unsafe_inline. This is the only other place I found a definitive answer: https://github.com/single-spa/create-single-spa/pull/44#issuecomment-608140480

Reply:
the fundamental functionality of importmap-rails is to make the browser do a bunch of ESM loading instead of bundling (eg, javascript files that contain `import` statements and reference other URLs). I think that means you need a working CSP policy for importmaps, and you probably didn't need one before because the JS files were bundled into a single named file loaded with a script tag and without invoking the browser's ESM loader.

Reply:
The importmap script tag gets evaluated as an unsafe-inline in selenium on localhost in test. I don't know why this works in dev but not in selenium, but it's refusing to even read the script tag.

unsafe-inline is ignored by browsers that support nonce. You can take our CSP and paste it [here](https://csp-evaluator.withgoogle.com) to see the recommended changes. It suggests adding unsafe inline for backwards compatibility with older browsers that don't support nonce.

CSP in this branch:
```
default-src 'self'; font-src 'self' https://fonts.gstatic.com; img-src 'self' https://secure.gaug.es https://gravatar.com https://www.gravatar.com https://secure.gravatar.com https://*.fastly-insights.com https://avatars.githubusercontent.com; object-src 'none'; script-src 'self' https://secure.gaug.es https://www.fastly-insights.com https://unpkg.com/@hotwired/stimulus/dist/stimulus.umd.js https://unpkg.com/stimulus-rails-nested-form/dist/stimulus-rails-nested-form.umd.js https://ga.jspm.io/npm:es-module-shims@1.8.2/dist/es-module-shims.js 'nonce-e2ad7278280e7bead9d9db2cd5c8ee17'; style-src 'self' https://fonts.googleapis.com 'nonce-e2ad7278280e7bead9d9db2cd5c8ee17'; connect-src 'self' https://s3-us-west-2.amazonaws.com/rubygems-dumps/ https://*.fastly-insights.com https://fastly-insights.com https://api.github.com http://localhost:*; form-action 'self' https://github.com/login/oauth/authorize; frame-ancestors 'self'; report-uri
```

Reply:
if there's a nonce, shouldn't it be used by the CSP?

Reply:
The result of this thread is that the nonce isn't always set on the first request because the session may not be updated during that request. Ultimately now we're adding the sha256 of the importmap to the CSP.